# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/*/ash/slakware-fix.patch
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./Makefile.orig	Sun May 15 14:16:31 1994
+++ ./Makefile	Tue Aug  3 16:19:10 1999
@@ -14,8 +14,10 @@
 
 OBJS =	$(OBJ1) $(OBJ2)
 
-CFLAGS = -O2 -fomit-frame-pointer -m486 -DSHELL -I/usr/include/bsd -I.
-LDFLAGS = -s -lbsd
+#CFLAGS = -O2 -pipe -DSHELL -I/usr/include/bsd -I. -D__BIT_TYPES_DEFINED__
+#LDFLAGS = -s -lbsd
+CFLAGS = -O2 -pipe -DSHELL -I. -D__BIT_TYPES_DEFINED__
+LDFLAGS = -s
 
 CLEANFILES =\
 	builtins.c builtins.h init.c mkinit mknodes mksyntax \
@@ -32,7 +34,7 @@
 token.def: mktokens
 	sh ./mktokens
 
-builtins.h builtins.c: mkbuiltins builtins
+builtins.h builtins.c: mkbuiltins
 	sh ./mkbuiltins
 
 init.c: mkinit $(SRCS)
--- ./mknodes.c.orig	Mon Sep  6 04:04:23 1993
+++ ./mknodes.c	Tue Aug  3 16:19:10 1999
@@ -89,7 +89,7 @@
 struct str *curstr;		/* current structure */
 
 
-FILE *infp = stdin;
+FILE *infp;
 char line[1024];
 int linno;
 char *linep;
@@ -102,6 +102,7 @@
 main(argc, argv)
 	char **argv;
 {
+	infp = stdin;
 	if (argc != 3) {
 		error("usage: mknodes file\n");
 		return(1);
--- ./trap.c.orig	Tue Aug 24 04:12:57 1993
+++ ./trap.c	Tue Aug  3 16:19:10 1999
@@ -53,6 +53,9 @@
 #include "mystring.h"
 #include <signal.h>
 
+#ifndef sig_t
+#define sig_t __sighandler_t
+#endif
 
 /*
  * Sigmode records the current value of the signal handlers for the various
